trl/trainer/dpo_trainer.py

```diff
@@ -1544,12 +1548,6 @@ def compute_loss(
         inputs: Dict[str, Union[torch.Tensor, Any]],
         return_outputs=False,
     ) -> Union[torch.Tensor, Tuple[torch.Tensor, Dict[str, torch.Tensor]]]:
-        if not self.use_dpo_data_collator:
-            warnings.warn(
-                "compute_loss is only implemented for DPODataCollatorWithPadding, and you passed a datacollator that is different than "
-                "DPODataCollatorWithPadding - you might see unexpected behavior. Alternatively, you can implement your own prediction_step method if you are using a custom data collator"
-            )
```

Comment:
For any warning, it should be possible for the user to perform an action that removes it. In this case, it's not possible. Therefore, I suggest deleting it.